CMakeLists.txt

```diff
@@ -145,6 +145,7 @@ include_directories("${CMAKE_CURRENT_SOURCE_DIR}/Libs/Common")
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/ExternalLibs/optparse")
 
 if(Build_PythonAPI)
+  SET(PYBIND11_PYTHON_VERSION 3.7)
```

Comment:
I don't understand it, but according [to this](https://github.com/pybind/pybind11/issues/587), this should be

```
set(PYBIND11_PYTHON_VERSION 3.7 CACHE STRING "")
```

Reply:
Caching is optional, so this isn't important.
I'm more interested in how it finds the correct version of Python compared to the main pybind branch.

Regarding the failure itself, that's because external python code can only be called from the main thread, and it's likely the googletest code uses separate threads. I've dealt with this in other projects by painstakingly creating a queue and executing those items with the main thread. I don't know how it could be done with googletest.

Reply:
I can't remember where we're getting googletest, but here's some muck about using it single threaded:
https://stackoverflow.com/questions/29335446/command-line-options-to-force-googletest-to-run-in-single-thread

Reply:
So... it finds the correct python because I fixed it awhile back to do that, which is why there is the set pybind11 python version, etc, etc. 

The failure to acquire the python gil seems to have been a problem from the very beginning. Was this work ever tested from Python?